src/Calibur.sol

```diff
@@ -190,6 +191,10 @@ contract Calibur is
         address to = _call.to == address(0) ? address(this) : _call.to;
 
         Settings settings = getKeySettings(keyHash);
+
+        // By default, only the root key or admin keys can self-call. This is to prevent untrusted keys from updating their own settings
+        // However, admin keys CAN update their own settings and evade the hook checks below
+        // To prevent this, we recommend adding a hook which disallows calls by admin keys to the register and `update` functions
```

Comment:
```suggestion
        // To prevent this, we recommend adding a hook which disallows calls by admin keys to the `register` and `update` functions
```